### PR TITLE
New service raw image uploader

### PR DESCRIPTION
**Pull request by jgleissner**
### What does this PR do? Why are we making this change?
This adds a service to upload raw image files to a specified location. Only supported job type at the moment supports uploading to an S3 bucket using the credentials from a specified AWS account.

### CODE REVIEW

File: test/unit/services/raw_image_uploader/config_test.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+from test.unit.test_helper import patch_open
+
+from mash.mash_exceptions import MashConfigException
+from mash.services.raw_image_uploader.config import RawImageUploaderConfig
+
+
+class TestRawImagerUploaderConfig(object):
+    def setup(self):
+        self.config = RawImageUploaderConfig(
+            '../data/mash_config.yaml'
```
Comment (smarlowucf) [inline test/unit/services/raw_image_uploader/config_test.py]: These should be updated to absolute path. `test/data/mash_config.yaml`

File: mash/services/raw_image_uploader_service.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    try:
+        logging.basicConfig()
+        log = logging.getLogger('MashService')
+        log.setLevel(logging.DEBUG)
+        # run service, enter main loop
+        ListenerService(
+            service_exchange='raw_image_uploader',
+            custom_args={
+                'listener_msg_args': ['cloud_image_name', 'image_file'],
+                'status_msg_args': ['cloud_image_name', 'source_regions']
```
Comment (smarlowucf) [inline mash/services/raw_image_uploader_service.py]: 'cloud_image_name' is already a default in `status_msg_args` https://github.com/SUSE-Enceladus/mash/blob/master/mash/services/listener_service.py#L70

File: mash/services/testing_service.py [hunk 3]
```diff
@@ -38,8 +38,8 @@ def main():
         ListenerService(
             service_exchange='testing',
             custom_args={
-                'listener_msg_args': ['cloud_image_name', 'source_regions'],
-                'status_msg_args': ['source_regions']
+                'listener_msg_args': ['cloud_image_name', 'image_file', 'source_regions'],
+                'status_msg_args': ['source_regions', 'image_file', 'cloud_image_name']
```
Comment (smarlowucf) [inline mash/services/testing_service.py]: Ditto on 'cloud_image_name'

File: test/unit/services/raw_image_uploader/config_test.py (hunk 1 above)
Comment (jgleissner) [reply test/unit/services/raw_image_uploader/config_test.py]: Sorry, I should have seen that change coming as I actually reviewed it. ;) PR rebased and updated.

File: mash/services/raw_image_uploader_service.py (hunk 2 above)
Comment (jgleissner) [reply mash/services/raw_image_uploader_service.py]: Fixed.

**Pull request merged**